iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/twin/TwinPnPTests.java

```diff
@@ -0,0 +1,185 @@
+    public static void tearDown() throws Exception
+    {
+        if (registryManager != null)
+        {
+            registryManager.close();
+            registryManager = null;
+        }
+    }
+
+    private void createDeviceWithClientOptions(IotHubClientProtocol protocol, ClientOptions options) throws IOException, URISyntaxException, InterruptedException
+    {
+        testInstance.testDevice.deviceClient = new DeviceClient(DeviceConnectionString.get(iotHubConnectionString, testInstance.deviceForRegistryManager), protocol, options);
+        testInstance.testDevice.deviceClient.open();
+    }
+
+    private void createModuleWithClientOptions(IotHubClientProtocol protocol, ClientOptions options) throws IOException, URISyntaxException, InterruptedException, ModuleClientException {
+        testInstance.testDevice.moduleClient = new ModuleClient(DeviceConnectionString.get(iotHubConnectionString, testInstance.deviceForRegistryManager, testInstance.moduleForRegistryManager), protocol, options);
+        testInstance.testDevice.moduleClient.open();
+    }
+
+    @Before
+    public void createDevice() throws Exception
+    {
+        testInstance.testDevice = new TestDevice();
+        testInstance.testDevice.deviceId = "java-twin-PnP-e2e-test-".concat(UUID.randomUUID().toString());
+        testInstance.testDevice.moduleId = "java-twin-PnP-module-e2e-test-".concat(UUID.randomUUID().toString());
+
+        testInstance.deviceForRegistryManager = com.microsoft.azure.sdk.iot.service.Device.createFromId(testInstance.testDevice.deviceId, null, null);
+        testInstance.deviceForRegistryManager = Tools.addDeviceWithRetry(registryManager, testInstance.deviceForRegistryManager);
+        testInstance.moduleForRegistryManager = com.microsoft.azure.sdk.iot.service.Module.createFromId(testInstance.testDevice.deviceId, testInstance.testDevice.moduleId,null);
+        testInstance.moduleForRegistryManager = Tools.addModuleWithRetry(registryManager, testInstance.moduleForRegistryManager);
+
+    }
+
+    @After
+    public void destroyDevice() throws Exception
+    {
+        if (testInstance.testDevice.deviceClient != null)
+        {
+            testInstance.testDevice.deviceClient.closeNow();
+            testInstance.testDevice.deviceClient = null;
+        }
+        if (testInstance.testDevice.moduleClient != null)
+        {
+            testInstance.testDevice.moduleClient.closeNow();
+            testInstance.testDevice.moduleClient = null;
+        }
+
+        if (testInstance != null && testInstance.testDevice != null)
+        {
+            if (registryManager != null && testInstance.testDevice.deviceId != null && testInstance.testDevice.moduleId !=null)
+            {
+                registryManager.removeModule(testInstance.testDevice.deviceId, testInstance.testDevice.moduleId);
+                registryManager.removeDevice(testInstance.testDevice.deviceId);
+            }
+        }
+    }
+
+    @Test
+    @StandardTierHubOnlyTest
```

Comment:
You can move this label up to the class level since all pnp twin tests will be standard tier only

Reply:
That's true I can move it up in this case

Reply:
Is there something like this for c# as well?

Reply:
Standard vs Basic tier? Not that I know of. C# doesn't test against basic tier hubs